tools/cryptogenotyper/cryptogenotyper.xml

```diff
@@ -0,0 +1,135 @@
+<tool id="CryptoGenotyper" name = "CryptoGenotyper" version = "1.0">
+  <description> CryptoGenotyper is a standalone tool to analyze <i>Cryptosporidium</i> Sanger sequencing data and classify the species and subtype based on SSU rRNA and <i>gp60</i> gene markers, respectively. </description>
+  <requirements>
+    <requirement type="package" version ="1.0">cryptogenotyper</requirement>
+  </requirements>
+  <command detect_errors="exit_code">
+     <![CDATA[
+
+     #set $ref_file=''
+     #set $ref_name=''
+
+
+     #if $reference.ref == "no"
+        ln -s "${reference.db}" "${reference.db.name}" &&
+        #set $ref_file = $reference.db.name
+     #end if
+
+     #if $primers["seqtype"] == "contig"
+        ln -s "${$primers.abi_input["forward"]}" "${primers.abi_input.name}_forward.ab1" &&
+        ln -s "${$primers.abi_input["reverse"]}" "${primers.abi_input.name}_reverse.ab1" &&
+        cryptogenotyper -i "." -m "$marker" -t "$primers.seqtype" -f "forward" -r "reverse"
+        #if $reference.ref == "no"
+          --databasefile $ref_file
+        #end if
+     #else
+        ln -s "${primers.abi_input}" "${primers.abi_input.element_identifier}" &&
+        cryptogenotyper -i "./${primers.abi_input.element_identifier}" -m "$marker" -t "$primers.seqtype" -f "" -r ""
+        #if $reference.ref == "no"
+          --databasefile $ref_file
+        #end if
+     #end if
+     #if $outputheader == "no"
+           --noheaderline
+     #end if
+     -o "result";
+     ]]>
+  </command>
+  <inputs>
+      <param name="marker" type="select" label="Marker">
+          <option value="18S">SSU rRNA</option>
+          <option value="gp60">gp60</option>
+      </param>
+      <conditional name="reference">
+        <param name="ref" type="select" label="Use default reference file?">
+            <option value="yes">Yes</option>
+            <option value="no">No</option>
+        </param>
+        <when value="no">
+            <param name="db" type="data" format="fasta" label="Reference Database File:" help=".fa fasta file type"/>
+        </when>
+      </conditional>
+      <conditional name="primers">
+        <param name="seqtype" type="select" label="Type of Sequences">
+            <option value="forward">Forward Only</option>
+            <option value="reverse">Reverse Only</option>
+            <option selected="true" value="contig">Contig</option>
```

Comment:
I'm not really familiar with this type of data, but is the appropriate word `contig` here or is it `read`? Here you use **Contig**, but elsewhere when talking about paired data you use **Read**.

Reply:
It is indeed a good observation. I had added a clarification in the description of the tool to indicate that contig mode related to forward and reverse paired reads. Apparently in the context of Cryptosporidium Sanger sequencing this terminology is used. Also in the manuscript it this paired-end mode is also referred as contig mode.  From what I understand given an amplicon it is possible to type Cryptosporidium based on forward primer (forward read) or reverse read (reverse primer) or using both forward and reverse reads (contig mode)

Reply:
Okay, thanks. That makes sense. I just wanted to make sure the terminology was consistent.